raiden/storage/sqlite.py

```diff
@@ -1472,16 +1473,18 @@ def get_latest_light_client_non_closing_balance_proof(self, channel_id, non_clos
                    balance_proof,
                    lc_balance_proof_signature
             FROM light_client_balance_proof
-            WHERE channel_id  = ? AND sender = ?
+            WHERE channel_id  = ? AND sender = ? and token_network_address = ?
             ORDER BY nonce DESC
             """,
-            (channel_id, to_checksum_address(non_closing_participant)),
+            (channel_id, to_checksum_address(non_closing_participant), to_checksum_address(token_network_address)),
         )
         return cursor.fetchone()
 
-    def delete_light_client_non_closing_balance_proof(self, channel_id):
+    def delete_light_client_non_closing_balance_proof(self, channel_id, token_network_addresss):
         with self.write_lock, self.conn:
-            self.conn.execute("DELETE FROM light_client_balance_proof WHERE channel_id = ?", (channel_id,))
+            self.conn.execute(
+                "DELETE FROM light_client_balance_proof WHERE channel_id = ? and token_network_address = ?",
+                (channel_id, to_checksum_address(token_network_addresss),))
```

Comment:
i don't understand the `,` after `addresss`, and the last `)` should be on a new line

Reply:
dont remember exactly why sqlite needs this but basically all the querys have it